apps/tmbitwarden.sh

```diff
@@ -79,38 +82,83 @@ Basically:
+else
+    userdel "$BITWARDEN_USER"
+    rm -rf /home/"$BITWARDEN_USER"
+fi
+groupadd docker >/dev/null 2>&1
+sudo usermod -aG docker "$BITWARDEN_USER"
+if [ ! -d /home/"$BITWARDEN_USER" ]
+then
+    mkdir -p /home/"$BITWARDEN_USER"
+    chown -R "$BITWARDEN_USER"n:docker /home/"$BITWARDEN_USER"
+fi
+
+# Create the service
+cat << BITWARDEN_SERVICE > /etc/systemd/system/bitwarden.service
+[Unit]
+Description=Bitwarden
+Requires=docker.service
+After=docker.service
+
+[Service]
+Type=oneshot
+User="$BITWARDEN_USER"
+Group="$BITWARDEN_USER"
+ExecStart=/home/"$BITWARDEN_USER"/bitwarden.sh start
+RemainAfterExit=true
+ExecStop=/home/"$BITWARDEN_USER"/bitwarden.sh stop
+
+[Install]
+WantedBy=multi-user.target
+BITWARDEN_SERVICE
+
+sudo chmod 644 /etc/systemd/system/bitwarden.service
+systemctl daemon-reload
+sudo systemctl enable bitwarden.service
+
 # Install Docker
 install_docker
 install_if_not docker-compose
 
 # Install Bitwarden 
 install_if_not curl
-cd /root
-curl_to_dir "https://raw.githubusercontent.com/bitwarden/core/master/scripts" "bitwarden.sh" "/root"
-chmod +x /root/bitwarden.sh
-check_command ./bitwarden.sh install
+cd /home/"$BITWARDEN_USER"
+curl_to_dir "https://raw.githubusercontent.com/bitwarden/core/master/scripts" "bitwarden.sh" "/home/$BITWARDEN_USER"
+chmod +x /home/"$BITWARDEN_USER"/bitwarden.sh
+check_command sudo -u "$BITWARDEN_USER" ./bitwarden.sh install
 
 # Check if all ssl settings were entered correctly
-if grep ^url /root/bwdata/config.yml | grep -q https || grep ^url /root/bwdata/config.yml | grep -q localhost
+if grep ^url /home/"$BITWARDEN_USER"/bwdata/config.yml | grep -q https || grep ^url /home/"$BITWARDEN_USER"/bwdata/config.yml | grep -q localhost
 then
     message "It seems like you have entered some wrong settings. We will remove bitwarden now again so that you can start over again."
     check_command ./bitwarden.sh install
     docker system prune -af
-    rm -rf /root/bwdata
+    rm -rf /home/"$BITWARDEN_USER"/bwdata
     exit 1
 fi
 
 # Continue with the installation
-sed -i "s|http_port.*|http_port: 5178|g" /root/bwdata/config.yml
-sed -i "s|https_port.*|https_port: 5179|g" /root/bwdata/config.yml
+sed -i "s|http_port.*|http_port: 5178|g" /home/"$BITWARDEN_USER"/bwdata/config.yml
+sed -i "s|https_port.*|https_port: 5179|g" /home/"$BITWARDEN_USER"/bwdata/config.yml
 # Get Subdomain from config.yml and change it to https
-SUBDOMAIN=$(grep ^url /root/bwdata/config.yml)
+SUBDOMAIN=$(grep ^url /home/"$BITWARDEN_USER"/bwdata/config.yml)
 SUBDOMAIN=${SUBDOMAIN##*url: http://}
-sed -i "s|^url: .*|url: https://$SUBDOMAIN|g" /root/bwdata/config.yml
-sed -i 's|http://|https://|g' /root/bwdata/env/global.override.env
-check_command ./bitwarden.sh rebuild
-check_command ./bitwarden.sh start
-check_command ./bitwarden.sh updatedb
+sed -i "s|^url: .*|url: https://$SUBDOMAIN|g" /home/"$BITWARDEN_USER"/bwdata/config.yml
+sed -i 's|http://|https://|g' /home/"$BITWARDEN_USER"/bwdata/env/global.override.env
```

Comment:
This is missing here:

sed -i "s|LOCAL_UID=.*|LOCAL_UID=$USERID|g" /home/"$BITWARDEN_USER"/bwdata/env/uid.env
sed -i "s|LOCAL_GID=.*|LOCAL_GID=$USERGROUPID|g" /home/"$BITWARDEN_USER"/bwdata/env/uid.env

Reply:
Aah, thanks. Just adding stuff as fast as possible so I can test now :D

Reply:
Please add btw :)

Reply:
done